oak_functions_launcher/src/lookup.rs

```diff
@@ -33,15 +33,18 @@ impl UpdateClient<'_> {
     async fn start(&mut self) -> anyhow::Result<()> {
         // Send the current chunk first.
         let update_response = self.send_request(UpdateAction::Start).await?;
-
         if UpdateStatus::Started != update_response.update_status() {
             return Err(anyhow!("Did not receive expected update status: Started"));
         };
+        self.continue_().await
+    }
 
-        // Currently send only data which fits in two chunks.
-        // TODO(#3718): to send more chunks.
-        if self.chunks.len() > 1 {
-            return Err(anyhow!("Cannot send lookup data with more than 2 chunks"));
+    async fn continue_(&mut self) -> anyhow::Result<()> {
+        while self.chunks.len() > 1 {
```

Comment:
I might be missing something here, but I want to take ownership of the elements of the iterator and  I think 
 `into_iter` would be the way. Or did I misunderstand something?

Reply:
`into_iter` is fine to use, but then you should use the result as an `Iter`, not as a `vec::IntoIter`, since in general the iterator may not come from a Vec at all. E.g. if it was created by streaming a table or some other data structure that cannot fit in memory (which is the common use case for using iterators instead of vecs), then the current approach would not really work. The advantage of an Iterator is that it hides the way that individual elements are produced.

Reply:
I think I got you now!